### ISSUE TITLE
MULTISET as JSON emulation using JSON_ARRAY must use NULL ON NULL clause

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12139]
Similar to #12136, when emulating `MULTISET` with `JSON`, we're not getting the correct results.

```java
Result<Record5<T, T, T, T, T>> r2 = ctx2.fetchValue(select(multiset(
    select(
        val(null, type).as("a"),
        val(value).as("b"),
        val(null, type).as("c"),
        val(value).as("d"),
        val(null, type).as("e"))
)));

assertEquals(1, r2.size());
assertEquals(5, r2.get(0).size());
assertNull(r2.get(0).value1());
assertEquals(forCompare.apply(value), r2.get(0).value2());
assertNull(r2.get(0).value3());
assertEquals(forCompare.apply(value), r2.get(0).value4());
assertNull(r2.get(0).value5());
```

Produces (in H2):

```sql
select (
  select coalesce(
    json_arrayagg(json_array("v0", "v1", "v2", "v3", "v4")),
    json_array(null on null)
  )
  from (
    select
      null "v0",
      'a' "v1",
      null "v2",
      'a' "v3",
      null "v4"
  ) "t"
)
```

Which ultimately deserialises as:

```
[(a, a, {null}, {null}, {null})]
```

The indexes are all off! This is due to a late switch from using `JSON_OBJECT()` to using `JSON_ARRAY()`, and the above test wasn't run for a few dialects. Will need to be fixed immediately for 3.16.0 and 3.15.1

**Comment by lukaseder** [comment]
Affected dialects include:

- H2
- Oracle

All the others default to `NULL ON NULL` behaviour.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0 and 3.15.1 (https://github.com/jOOQ/jOOQ/issues/12140)
